datashader/bundling.py

```diff
@@ -77,96 +106,127 @@ def resample_segment(segments, new_segments, min_segment_length, max_segment_len
     new_segments[pos] = next_point
     return new_segments
 
-
-@ngjit
-def calculate_length(segments, min_segment_length, max_segment_length):
+@nb.jit(
+    nb.types.Tuple((nb.boolean, nb.uint64, nb.uint16[::1]))(nb.float32[:,::1], segment_length_type),
+    nopython=True,
+    nogil=True,
+    fastmath=True,
+    locals={
+        'next_point': nb.float32[::1],
+        'current_point': nb.float32[::1],
+        'pos': nb.uint64,
+        'index': nb.uint64,
+        'distance': nb.float32
+    }
+)
+def calculate_resampling(segments, squared_segment_length):
     current_point = segments[0]
     index = 1
     total = 0
     any_change = False
+    n_points_to_add = np.zeros(len(segments), dtype=np.uint16)
     while index < len(segments):
         next_point = segments[index]
         distance = distance_between(current_point, next_point)
-        if (distance < min_segment_length and 1 < index < (len(segments) - 2)):
+        if (distance < squared_segment_length.min and 1 < index < (len(segments) - 2)):
+            # Merge points
             any_change = True
             current_point = (current_point + next_point) / 2
+            n_points_to_add[index] = 0
             total += 1
             index += 2
-        elif distance > max_segment_length:
+        elif distance > squared_segment_length.max:
             any_change = True
             # Linear subsample
-            points = int(ceil(distance / ((max_segment_length + min_segment_length) / 2)))
+            points = np.uint16(ceil(np.sqrt(distance / squared_segment_length.mean)))
+            n_points_to_add[index] = points
             total += points
             current_point = next_point
             index += 1
         else:
             # Do nothing
+            n_points_to_add[index] = 1
             total += 1
             current_point = next_point
             index += 1
     total += 1
-    return any_change, total
-
-
-def resample_edge(segments, min_segment_length, max_segment_length, ndims):
-    change, total_resamples = calculate_length(segments, min_segment_length, max_segment_length)
+    return any_change, total, n_points_to_add
+
+@nb.jit(
+    nb.float32[:,::1](nb.float32[:,::1], segment_length_type, nb.int64),
```

Comment:
This is a good example of why we may want to include formatting rules

```suggestion
    nb.float32[:, ::1](nb.float32[:, ::1], segment_length_type, nb.int64),
```